nsqadmin/logger.go

```diff
@@ -1,5 +1,58 @@
 package nsqadmin
 
+import (
+	"fmt"
+	"strings"
+)
+
 type Logger interface {
 	Output(maxdepth int, s string) error
 }
+
+const (
+	LOG_DEBUG = 1
+	LOG_INFO  = 2
+	LOG_WARN  = 3
+	LOG_ERROR = 4
+	LOG_FATAL = 5
+)
+
+func (n *NSQAdmin) logLevelFromString(level string) int {
+	// check log-level is valid and translate to int
+	switch strings.ToLower(level) {
+	case "debug":
+		return LOG_DEBUG
+	case "info":
+		return LOG_INFO
+	case "warn":
+		return LOG_WARN
+	case "error":
+		return LOG_ERROR
+	case "fatal":
+		return LOG_FATAL
+	case "":
+		return LOG_INFO
+	default:
+		return -1
+	}
+}
+
+func (n *NSQAdmin) logf(level int, f string, args ...interface{}) {
+	levelString := "INFO"
+	switch level {
+	case LOG_DEBUG:
+		levelString = "DEBUG"
+	case LOG_INFO:
+		levelString = "INFO"
+	case LOG_WARN:
+		levelString = "WARNING"
+	case LOG_ERROR:
+		levelString = "ERROR"
+	case LOG_FATAL:
+		levelString = "FATAL"
+	}
+
+	if level >= n.getOpts().logLevel || n.getOpts().Verbose {
+		n.getOpts().Logger.Output(2, fmt.Sprintf(levelString+": "+f, args...))
```

Comment:
How about you do `opts = n.GetOpts()` just once above, so the `atomic.Value` is fetched and cast just once.